enumerable.rb

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env ruby
+
+# Some methods will be added to this module
+module Enumerable # rubocop:disable Metrics/ModuleLength
+  # definition of the my_each method
+
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    for element in self do # rubocop:disable Style/For
+      yield(element)
+    end
+    self
+  end
+
+  # definition of the my_each_with_index method
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    index = 0
+    my_each do |element|
+      yield(element, index)
+      index += 1
+    end
+    self
+  end
+
+  # definition of the my_select method
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_array = []
+    my_each do |element|
+      new_array << element if yield(element)
+    end
+    new_array
+  end
+
+  # definition of the my_all? method
+
+  def my_all?(pattern = nil) # rubocop:disable CyclomaticComplexity, PerceivedComplexity
```

Comment:
* [ ] should have working `my_all?` **(covered by automated tests)** identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
Let us fix this
 ```p [nil, true, 99].my_all? != p [nil, true, 99].all?```